src/provider/states/create_config.rs

```diff
@@ -0,0 +1,296 @@
+use std::collections::BTreeMap;
+use std::fs;
+use std::fs::read_to_string;
+use std::path::PathBuf;
+
+use handlebars::Handlebars;
+use k8s_openapi::api::core::v1::ConfigMap;
+use kube::{Api, Client};
+use kubelet::pod::Pod;
+use kubelet::state::prelude::*;
+use kubelet::state::{State, Transition};
+use log::{debug, error, info, trace, warn};
+
+use crate::fail_fatal;
+use crate::provider::error::StackableError;
+use crate::provider::error::StackableError::PodValidationError;
+use crate::provider::states::create_service::CreatingService;
+use crate::provider::states::setup_failed::SetupFailed;
+use crate::provider::states::waiting_config::WaitingConfigMap;
+use crate::provider::PodState;
+
+#[derive(Default, Debug, TransitionTo)]
+#[transition_to(CreatingService, SetupFailed, WaitingConfigMap)]
+pub struct CreatingConfig {
+    pub target_directory: Option<PathBuf>,
+}
+
+impl CreatingConfig {
+    pub fn render_config_template(
+        data: BTreeMap<String, String>,
+        template: String,
+    ) -> Result<String, StackableError> {
+        let mut handlebars = Handlebars::new();
+        debug!("Rendering template with context: {:?}", data);
+
+        // register the template. The template string will be verified and compiled.
+        handlebars.register_template_string("t1", template)?;
+
+        // Set strict mode, so that we fail with an error if any non-existent fields are accessed
+        handlebars.set_strict_mode(true);
+
+        // Render the template with the provided data and return the resulting String
+        Ok(handlebars.render("t1", &data)?)
+    }
+
+    pub fn create_render_data(pod_state: &PodState) -> BTreeMap<String, String> {
+        let mut render_data = BTreeMap::new();
+        let directory_name = pod_state.package.get_directory_name();
+
+        if let Ok(package_dir) = &pod_state
+            .get_service_package_directory()
+            .into_os_string()
+            .into_string()
+        {
+            render_data.insert(String::from("packageroot"), String::from(package_dir));
+        } else {
+            warn!("Unable to parse value for package directory as UTF8")
```

Comment:
shouldn't this abort instead? 

let package_dir = ......into_string()?;

Reply:
I've refactored this to return an error and abort instead.